ingest-app/src/cmr/ingest/api/generic_documents.clj

```diff
@@ -142,8 +142,35 @@
     (validate-document-against-schema spec-key spec-version metadata)
     ;; The update-generic is a macro which allows for a list of URL parameters to be
     ;; passed in to be resolved by a function.
-    (tgen/update-generic request-context [provider-id native-id] (json/generate-string concept))
-    {:status 204}))
+    (tgen/update-generic request-context [provider-id native-id] (json/generate-string concept))))
 
 (defn delete-generic-document [request]
   (println "stub function: delete " request))
+
+;; TODO: Generic work: This could be a candidate for a configuration file.
+(defn validate-and-get-required-parameters
```

Comment:
this only ever returns one parameter, the provider object (which I think is suppose to include the other provider fields but which I never needed in any code so I faked it by creating a map with only provider-id). I think you should return the provider object in a map under the key :provider (or something like that name) so that you can add new ones. Also, native-id is required right?

Reply:
Right now all this function does is grab a query parameter named :provider and checks to see if the provider exists.
/search/orderoption?provider=PROV1.  
An exception is thrown if the provider does not exist otherwise it returns a map of key (provider and the query parameter value.

It doesn't do anything with the parameter other than to check that it exists on the query parameter.  It doesn't get the provider object and it doesn't do anything with it.
Then it just returns a map of the key and the string name of the provider that it got from the query parameters. 

If you don't have native id in the url, then you won't get this far in the code.